tests/unit_tests/core/router/identity.cc

```diff
@@ -85,4 +86,47 @@ BOOST_AUTO_TEST_CASE(ParseIdentityFailure)
         0);
 }
 
+BOOST_AUTO_TEST_CASE(ValidCreateRoutingKey)
+{
+  core::IdentityEx ident;
+  BOOST_CHECK(
+      ident.FromBuffer(m_AliceIdentity.data(), m_AliceIdentity.size()));
+  BOOST_CHECK_NO_THROW(core::CreateRoutingKey(ident.GetIdentHash()));
+  // Ensure a valid identity hash is returned
+  BOOST_CHECK(core::CreateRoutingKey(ident.GetIdentHash()).ToBase32().size());
+}
+
+BOOST_AUTO_TEST_CASE(CreateInvalidRoutingKey)
+{
+  kovri::core::IdentHash zeroed_ident;
+  BOOST_CHECK_THROW(core::CreateRoutingKey(nullptr), std::invalid_argument);
+  BOOST_CHECK_THROW(core::CreateRoutingKey(zeroed_ident), std::invalid_argument);
+}
+
+BOOST_AUTO_TEST_CASE(ValidFormatRoutingKey)
+{
+  core::IdentityEx ident;
+  BOOST_CHECK(
+      ident.FromBuffer(m_AliceIdentity.data(), m_AliceIdentity.size()));
+
+  BOOST_CHECK_NO_THROW(core::FormatRoutingKey(ident.GetIdentHash()));
+  const std::string formatted_key = core::FormatRoutingKey(ident.GetIdentHash());
+  const core::IdentHash formatted_ident(reinterpret_cast<const std::uint8_t*>(
+      formatted_key.substr(0, ident.GetIdentHash().size()).data()));
```

Comment:
> const std::string formatted_key = core::FormatRoutingKey(ident.GetIdentHash());
>const core::IdentHash formatted_ident(reinterpret_cast<const std::uint8_t*>(
>    formatted_key.substr(0, ident.GetIdentHash().size()).data()));

This block is logically related to the comment below, not the check above. For sake of readability, you should put this block *below* the line *// Ensure formatted routing key contains the original ident hash*.

Reply:
Ok, I must have missed that when making changes to the tests. I'll fix it.